docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+  
+  On the other hand, Pipeline API allows developers to construct and execute pipelines with multiple neural network models, multiple input and output nodes, multiple data processors, pre-and-post processors, and various data path manipulators. Besides, if the input is online data (or streamed data), Pipeline APIs simplify your application and improve its performance.
+  
+- **Support various neural net frameworks (NNFW)**
+  
+  TensorFlow, TensorFlow-Lite, Caffe2, and PyTorch are support. Thus, neural network model files trained by such frameworks can be imported as filters of pipelines directly.
+  Custom filters, which are neural network models implemented directly with programming languages including C/C++ and Python, may be imported as filters of pipelines directly as well.
+
+  Note that, in Tizen release, this is limited to TensorFlow-Lite only; we do not ship other frameworks for official embedded device OS images.
+
+## Prerequisite <a name="prerequisite"></a>
+
+To use the APIs, below headers and features are required.
+
+- **Header**
+  ```bash
+  <nnstreamer.h>
+  ```
+
+- **Feature**
+  ```bash
+  <feature name="http://tizen.org/feature/machine_learning">true</feature>
+  <feature name="http://tizen.org/feature/machine_learning.inference">true</feature>
+  ```
+
+
+## Single API <a name="singleapi"></a>
+
+1. Open a model file with Single API
+    ```bash
+      <nnstreamer-single.h>
+      
+      ml_single_h single;
+      ml_tensors_info_h in_info, out_info;
+      ...
+
+      ml_single_open (&single, "model_file_path", in_info, out_info, ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY);
+    ```
+    
+    To load a model file, 2 `ml_tensors_info_h` are required. `in_info` contains the information of input tensor, and `out_info` is for output tensor. For the detail, you can check [Tensors Info section](#tensorsinfo).
+
+2. APIs about [Tensor info](#tensorsinfo).
+    After model load, below APIs bring the info of input & output tensors.
+    ```bash  
+      ml_single_get_input_info (ml_single_h single, ml_tensors_info_h * info)
+      ml_single_get_output_info (ml_single_h single, ml_tensors_info_h * info)
+    ```
+
+3. Invoke the model with input & output [Tensor data](#tensorsdata)
+    The model can be invoked with input & output tensor data. The result is included at the output tensor data.
+    ```bash
+      ml_tensors_data_create (in_info, &input);
+      ml_single_invoke (single, input, &output);
+    ```
+
+4. close the opened handle
+    ```bash
+    ml_single_close (single);
+    ```
+
+## Pipeline API <a name="pipelineapi"></a>
```

Comment:
As a best practice, it would be good to give a small introduction line before the steps begin.